ocs/agents/influxdb_publisher_v2/agent.py

```diff
@@ -0,0 +1,167 @@
+
+    def _enqueue_incoming_data(self, _data):
+        """Data handler for all feeds. This checks to see if the feeds should
+        be recorded, and if they are it puts them into the incoming_data queue
+        to be processed by the Publisher during the next run iteration.
+
+        """
+        data, feed = _data
+
+        if not feed['record'] or not self.aggregate:
+            return
+
+        self.incoming_data.put((data, feed))
+
+    @ocs_agent.param('test_mode', default=False, type=bool)
+    def record(self, session: ocs_agent.OpSession, params):
+        """record()
+
+        **Process** - This process will create an Publisher instance, which
+        will collect and write provider data to disk as long as this process is
+        running.
+
+        Parameters:
+            test_mode (bool, optional): Run the record Process loop only once.
+                This is meant only for testing. Default is False.
+
+        """
+        self.aggregate = True
+
+        self.log.debug("Instatiating Publisher class")
+        publisher = Publisher(self.args.database,
+                              self.incoming_data,
+                              self.args.org,
+                              protocol=self.args.protocol,
+                              gzip=self.args.gzip,
+                              operate_callback=lambda: self.aggregate,
+                              )
+
+        session.set_status('running')
+        while self.aggregate:
+            time.sleep(self.loop_time)
+            self.log.debug(f"Approx. queue size: {self.incoming_data.qsize()}")
+            publisher.run()
+
+            if params['test_mode']:
+                break
+
+        publisher.close()
+
+        return True, "Aggregation has ended"
+
+    def _stop_record(self, session, params):
+        if OpCode(session.op_code) in [OpCode.STARTING, OpCode.RUNNING]:
+            session.set_status('stopping')
+            self.aggregate = False
+            return True, "Stopping aggregation"
+        elif OpCode(session.op_code) == OpCode.STOPPING:
+            return True, "record process status is already 'stopping'"
+        else:
+            return False, "record process not currently running"
+
+
+def make_parser(parser=None):
+    if parser is None:
+        parser = argparse.ArgumentParser()
+
+    pgroup = parser.add_argument_group('Agent Options')
+    pgroup.add_argument('--initial-state',
+                        default='record', choices=['idle', 'record'],
+                        help="Initial state of argument parser. Can be either "
+                             "idle or record")
+    pgroup.add_argument('--database',
+                        default='ocs_feeds',
+                        help="Database within InfluxDB to publish data to.")
+    pgroup.add_argument('--org',
+                        default='my-org',
+                        help="Organization within InfluxDB to publish data to.")
```

Comment:
Related to my comment about reading from the `.env` in `drivers.py`, this would end up getting dropped.